src/types/predicates.ts

```diff
@@ -15,7 +15,7 @@ export type ObjectPrototypeKeys = keyof Object;
 /** no-doc */
 export type FunctionPrototypeKeys = keyof Function;
 
-export type IsNever<S extends string> = Not<(Record<S, True> & Record<string, False>)[S]>;
+export type IsNever<T> = keyof any extends keyof T ? Not<IsAny<T>> : False;
```

Comment:
It's weird to me that `keyof never` is the same as `keyof any`, while `keyof unknown` is `never`.  This feels like the sort of thing that might break in a later version of TS, but I suppose that's what the tests are for...